### PR TITLE
Supports course_ids

### PR DESCRIPTION
**Pull request by nfischer**
This allows support for multiple courses via the use of course_ids. This means that the URL specified must have the course_id as part of it, so this is an API change. This change is implemented on both frontend and backend, so this shouldn't break anything.

Also, I caught a bug that I didn't catch in my last pull request, so this fixes that issue. On the current version of master, adding nodes through the frontend will fail during the "update parent node's children" step. This resolves the issue and also deprecates one of the functions in the frontend.

This uses course_id "42" by default (it's hard coded for now). This number can be changed if you change it in the frontend (where it should change dynamically depending on which course is looked at) as well as in `addSampleData.py` script (which is temporary anyway).

### CODE REVIEW

File: frontend/src/utils/webapi.js [hunk 1]
```diff
@@ -8,10 +8,13 @@ import Node from '../Models/node.js';
...
 //----------------------------------------
 
+// TODO(nate): This is a hardcoded <course_id>. Change this dynamically during
+// runtime based on which course we're actually viewing
+var course_id = "42";
 var mainUrl = "";
 
 function getNode(nodeId: string){
-  let endpoint = mainUrl + `/node/${nodeId}/`;
+  let endpoint = mainUrl + '/' + course_id + `/node/${nodeId}/`;
```
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: these should use template strings as was done before

Comment (sakekasi) [inline frontend/src/utils/webapi.js]: `/${course_id}/node/${nodeId}/`

Comment (sakekasi) [inline frontend/src/utils/webapi.js]: also, this is just nitpicking but matching the current camelCase would be good for consistency in variable names

File: frontend/src/utils/webapi.js [hunk 2]
```diff
@@ -19,16 +22,18 @@ function getNode(nodeId: string){
 }
 
 function overwriteNode(node: Node){
-  let endpoint = mainUrl + `/node/update/${node.id}/`;
+  let endpoint = mainUrl + '/' + course_id + `/node/update/${node.id}/`;
```
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: see previous template string comment

File: frontend/src/utils/webapi.js [hunk 3]
```diff
@@ -19,16 +22,18 @@ function getNode(nodeId: string){
...
-    data: node,
+    data: data,
     dataType: "json"
   });
 }
 
+// @deprecated
 function overwriteChildren(node: Node){
-  let endpoint = mainUrl + `/node/update/${node.id}/`;
+  let endpoint = mainUrl + '/' + course_id + `/node/update/${node.id}/`;
```
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: see previous template string comment

File: frontend/src/utils/webapi.js [hunk 4]
```diff
@@ -38,7 +43,7 @@ function overwriteChildren(node: Node){
 }
 
 function createNode(node: Node){ //mock for creation process
-  let endpoint = mainUrl + "/node/add/";
+  let endpoint = mainUrl + "/" + course_id + "/node/add/";
```
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: see previous template string comment

File: frontend/src/utils/webapi.js [hunk 5]
```diff
@@ -48,7 +53,7 @@ function createNode(node: Node){ //mock for creation process
 
 //currently rootId and depth are ignored
 function getTree(rootId = null, depth = null){
-  let endpoint = mainUrl + "/tree/";
+  let endpoint = mainUrl + "/" + course_id  + "/tree/";
```
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: see previous template string comment

File: frontend/src/utils/webapi.js (hunk 1 above)
Comment (sakekasi) [inline frontend/src/utils/webapi.js]: BTW template strings are delimited by backticks as seen in the current code

Comment (nfischer) [inline frontend/src/utils/webapi.js]: Thanks for the review. Forgot to check naming consistency. I'll update this when I get home

Comment (sakekasi) [inline frontend/src/utils/webapi.js]: sounds good. other than this, everything looks good. will pull once addressed

**Comment by nfischer** [comment]
I think this should be ready to merge now

**Pull request merged**